pkg/keccak/contrib/Modes/keccak1600.h

```diff
@@ -0,0 +1,169 @@
+/*
+
+Implementation by the Keccak, Keyak and Ketje Teams, namely, Guido Bertoni,
+Joan Daemen, Michaël Peeters, Gilles Van Assche and Ronny Van Keer, hereby
+denoted as "the implementer".
+
+For more information, feedback or questions, please refer to our websites:
+http://keccak.noekeon.org/
+http://keyak.noekeon.org/
+http://ketje.noekeon.org/
+
+To the extent possible under law, the implementer has waived all copyright
+and related or neighboring rights to the source code in this file.
+http://creativecommons.org/publicdomain/zero/1.0/
+
+This is free and unencumbered software released into the public domain.
+
+Anyone is free to copy, modify, publish, use, compile, sell, or
+distribute this software, either in source code form or as a compiled
+binary, for any purpose, commercial or non-commercial, and by any
+means.
+
+In jurisdictions that recognize copyright laws, the author or authors
+of this software dedicate any and all copyright interest in the
+software to the public domain. We make this dedication for the benefit
+of the public at large and to the detriment of our heirs and
+successors. We intend this dedication to be an overt act of
+relinquishment in perpetuity of all present and future rights to this
+software under copyright law.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+OTHER DEALINGS IN THE SOFTWARE.
+
+For more information, please refer to <http://unlicense.org/>
+*/
+
+#ifndef KECCAK1600_H
+#define KECCAK1600_H
+
+#define KeccakP200_excluded
+#define KeccakP400_excluded
+
+#ifndef KeccakP1600_excluded
+
+#include "keccak_defs.h"
+#include "KeccakHash.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define keccak1600hash_instance Keccak_HashInstance
+#define fixed_output_length     fixedOutputLength
+#define delimited_suffix        delimitedSuffix
```

Comment:
Doc missing for those macros. Also not sure if they are necessary... (I know I said make this file compliant to our coding conventions, but this is a bit excessive ;-)).

Reply:
They are necessary if we want the API for Keccak-1600 to have a naming scheme consistent to Keccak-800.

These are macros for the keccak1600hash_instance struct and two of the variables inside. Should I add documentation for the analogous struct keccak800hash_instance in keccak800.h as well?